avocado/core/multiplexer.py

```diff
@@ -377,28 +377,95 @@ def iteritems(self):
                 yield (leaf.environment_origin[key].path, key, value)
 
 
+def _report_mux_already_parsed(self, *args, **kwargs):
+    """
+    Raises exception describing that `self.data` alteration is restricted
+    """
+    raise RuntimeError("Mux already parsed, altering is restricted. %s %s"
+                       % (args, kwargs))
+
+
 class Mux(object):
 
     """
     This is a multiplex object which multiplexes the test_suite.
     """
 
-    def __init__(self, args):
+    def __init__(self):
         self._has_multiple_variants = None
-        mux_files = getattr(args, 'multiplex_files', None)
+        self.variants = None
+        self.data = tree.TreeNode()
+        self._mux_path = None
+
+    def parse(self, args):
+        """
+        Apply options defined on the cmdline
+
+        :param args: Parsed cmdline arguments
+        """
         filter_only = getattr(args, 'filter_only', None)
         filter_out = getattr(args, 'filter_out', None)
-        if mux_files:
-            mux_tree = yaml2tree(mux_files)
-        else:   # no variants
-            mux_tree = tree.TreeNode()
-        if getattr(args, 'default_avocado_params', None):
-            mux_tree.merge(args.default_avocado_params)
-        mux_tree = tree.apply_filters(mux_tree, filter_only, filter_out)
+        self._parse_basic_injects(args)
+        mux_tree = tree.apply_filters(self.data, filter_only, filter_out)
         self.variants = MuxTree(mux_tree)
         self._mux_path = getattr(args, 'mux_path', None)
         if self._mux_path is None:
             self._mux_path = ['/run/*']
+        # disable data alteration (and remove data as they are not useful)
+        self.data = None
+        self.data_inject = _report_mux_already_parsed
+        self.data_merge = _report_mux_already_parsed
+
+    def _parse_basic_injects(self, args):
+        """
+        Inject data from the basic injects defined by Mux
+
+        :param args: Parsed cmdline arguments
+        """
+        # Merge the multiplex_files
+        multiplex_files = getattr(args, "multiplex_files", None)
+        if multiplex_files:
+            self.data_merge(yaml2tree(multiplex_files))
+
+        # FIXME: Backward compatibility params, to be removed when 36 LTS is
+        # discontinued
+        if (not getattr(args, "mux_skip_defaults", False) and
+                hasattr(args, "default_avocado_params")):
+            self.data_merge(args.default_avocado_params)
```

Comment:
Oh, now I see that compatibility with `default_avocado_params` is also performed here.